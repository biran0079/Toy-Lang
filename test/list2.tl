a=[];
i=0;
while(i<5){
  a = a+i;
  i = i+1;
}
print(a);
a=a+[3,2,1];
print(a);
b=[1, 2, 3];
b=b+b;
print(b);
c=[0];
c=c+[[1,2]];
print(c);
print(c[1][1]);
print(len(c));
