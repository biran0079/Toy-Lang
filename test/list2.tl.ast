[stmts, [=, [id, a], [list]], [=, [id, i], [int, 0]], [while, [<, [id, i], [int, 5]], [stmts, [=, [id, a], [+, [id, a], [list, [id, i]]]], [=, [id, i], [+, [id, i], [int, 1]]]]], [call, [id, print], [exps, [id, a]]], [=, [id, a], [+, [id, a], [list, [int, 3], [int, 2], [int, 1]]]], [call, [id, print], [exps, [id, a]]], [=, [id, b], [list, [int, 1], [int, 2], [int, 3]]], [=, [id, b], [+, [id, b], [id, b]]], [call, [id, print], [exps, [id, b]]], [=, [id, c], [list, [int, 0]]], [=, [id, c], [+, [id, c], [list, [list, [int, 1], [int, 2]]]]], [call, [id, print], [exps, [id, c]]], [call, [id, print], [exps, [list_access, [list_access, [id, c], [int, 1]], [int, 1]]]], [call, [id, print], [exps, [call, [id, len], [exps, [id, c]]]]]]
