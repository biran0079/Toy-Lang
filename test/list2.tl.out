[0, 1, 2, 3, 4]
[0, 1, 2, 3, 4, 3, 2, 1]
[1, 2, 3, 1, 2, 3]
[0, [1, 2]]
2
2
